pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -912,6 +944,17 @@ func proposeBatch(raftGroup proposerRaft, replID roachpb.ReplicaID, ents []raftp
 		// ignored prior to the introduction of ErrProposalDropped).
```

Comment:
It's still true, better handling of it would at least entail adding metrics, etc.
I think other than that it should already work about as well as it can, we'll repropose everything when we get a `reasonNewLeaderOrConfChange` refresh, but I can't say this has exactly been verified 😄 so I think this TODO should stay for now.